crates/solver/src/solver/zeroex_solver.rs

```diff
@@ -154,57 +153,18 @@ impl SingleOrderSolving for ZeroExSolver {
             .await
             .context("get_approval")?;
 
-        let create_settlement = || {
```

Comment:
I dropped this logic and its test since the `SettlementRanker` already discards solutions with negative scores.